feat(alerts): alerts button on top of insight

anirudhpillai opened the pull request:
## Problem
Hard to discover that you can set alerts on insight

## Changes
Added a button to manage alerts right on top of the insight so users can easily access it. Can also show count of alerts set in the button but first wanted to check people are happy with the UX change.

Note: I've added it as part of the 'caption' of the page header. Thought this was cleaner as the user activity indicator doesn't need the full space.

<img width="836" alt="image" src="https://github.com/user-attachments/assets/33c7c167-9819-453d-b5d3-3f0b4af6285a">

## Does this work well for both Cloud and self-hosted?
N/A
<!-- Yes / no / it doesn't have an impact. -->

## How did you test this code?
Tested locally
<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

anirudhpillai committed:
feat(alerts): alerts button on top of insight

thmsobrmlr reviewed (changes requested):
> Can also show count of alerts set in the button ...

That would be great! Can of course come later as well.

> ... but first wanted to check people are happy with the UX change.

Fine for me to go in here - should probably be size "small" then to fit in with the other buttons at the top. Another suggestion would be to put it next to the "detailled results" table next to the export button. After all exports and alerts are similar in that they get data out.

A bit of nit-picking:
- Going to "edit mode" makes the button disappear, which is strange. The button therefore also disappears when the alerts modal is open. Maybe you want to check if this is a persisted insight (`hasDashboardItemId`)?
- The text for the disabled reason says "Insights are only available for trends without breakdowns". Shouldn't it be "**Alerts** are only available for trends without breakdowns."
- Tests are failing somehow.

anirudhpillai committed:
move alerts button to top bar

anirudhpillai committed:
update tests

anirudhpillai commented:
Thanks Thomas! Have moved the button to the top header. I wanted it there originally but was hesitant in case it uses it up too much space, though seems good now.
Also added the icon to indicate number of alerts setup
<img width="842" alt="image" src="https://github.com/user-attachments/assets/8cc4b079-0d7c-44ef-a76b-2c0ac873318a">

Thanks for checking, have also fixed the warning text
<img width="472" alt="image" src="https://github.com/user-attachments/assets/73805a14-e29f-4975-9805-a27f464ff6dc">

Have also updated the tests, was waiting to update the reference to the button after the UX is approved :) 

Just regarding `Going to "edit mode" makes the button disappear`. I had done this intentionally as users can change the insight while editing so that alerts aren't supported anymore. Now it's also consistent with the other buttons in the top bar which also get hidden in edit mode
<img width="844" alt="image" src="https://github.com/user-attachments/assets/0ed7da59-7fd1-408e-a570-07007883b99a">

anirudhpillai committed:
update tests

thmsobrmlr reviewed (approved):
Nice work! Two things, but they’re just observations—no need to take any action on them.

1. We could remove the "Add to..." text from notebooks and dashboards as well. Would be more uniform and take up less space:
<img width="379" alt="Screenshot 2024-11-13 at 10 09 48" src="https://github.com/user-attachments/assets/64ef55bf-583b-40c2-bfd7-e6c6158493fd">

2. When I add an alert and then change the insight type to e.g. funnel, the button looks like the following. I don't know how the alert will work now and can't edit it:
<img width="436" alt="Screenshot 2024-11-13 at 10 10 18" src="https://github.com/user-attachments/assets/20a85284-6db3-4351-8873-c484e0a591cf">
